### ISSUE TITLE
Locale negotiation support

### ISSUE THREAD

**Issue by agseco** [mashpie/i18n-node#296]
Hi,

I have not found support for locale negotiation in this library, isn't it supported or I just could not find it? I could not find any issue referring to this topic either.

Assuming it's not supported, it would be fantastic if the module were able to resolve the effective locale given the list of supported locales and the requested locale.

For the moment I did a workaround using this library: https://github.com/florrain/locale

Cheers.

**Comment by falxcerebri** [comment]
Yeah! I'm totally stuck while trying to combine https://github.com/florrain/locale with i18n. 

The problem could be resolved perhaps by allowing user to overwrite `guessLanguage` i18n function or overwriting i18n.init in your app (that's way more work, you need to implement `applyAPItoObject` yourself and some other functions).

**Comment by falxcerebri** [comment]
FYI: I've done this in 5 minutes using this: https://github.com/jeresig/i18n-node-2 
```
[...]
var i18n = require("i18n-2");
[...]
i18n.expressBind(app, {
    locales: ['en', 'pl'],
    extension: ".json"
});

app.use(locale({
  priority: ["cookie", "accept-language", "default"],
  default: "en_US",
  cookie: {name: 'locale'}
}));

app.use(function(req, res, next) {
    req.i18n.setLocale(req.locale.language);
    next();
});
[...]
```
